sdk/modelsrepository/azure-iot-modelsrepository/src/main/java/com/azure/iot/modelsrepository/implementation/DtmiConventions.java

```diff
@@ -0,0 +1,49 @@
+package com.azure.iot.modelsrepository.implementation;
+
+import java.util.regex.Pattern;
+
+public class DtmiConventions {
+
+    private static final Pattern validDtmiPattern = Pattern.compile("^dtmi:[A-Za-z](?:[A-Za-z0-9_]*[A-Za-z0-9])?(?::[A-Za-z](?:[A-Za-z0-9_]*[A-Za-z0-9])?)*;[1-9][0-9]{0,8}$");
+
+    public static boolean isValidDtmi(String dtmi) {
```

Comment:
Is this a helper function for users, or will we be using this to validate DTMI's for the user in some way? Will the service be able to validate this?

Reply:
There is no service involved, just a CDN in front of a storage account, so the responsibility falls into the SDKs hands.
The plan is to ship this as a helper class to users as well. I will move it away from the implementation package.